### PR TITLE
hono-2515: Add LoRa provider for the things stack

### PR DESCRIPTION
**Pull request by BobClaerhout**
https://github.com/eclipse/hono/issues/2515
Signed-off-by: Bob Claerhout <claerhout.bob@gmail.com>

### CODE REVIEW

File: adapters/lora-vertx/src/main/java/org/eclipse/hono/adapter/lora/providers/TheThingsStackProvider.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    private static final String FIELD_THE_THINGS_STACK_SPREADING_FACTOR = "spreading_factor";
+    private static final String FIELD_THE_THINGS_STACK_UPLINK = "uplink_message";
+    private static final String FIELD_THE_THINGS_STACK_USER = "user";
+
+    private static final String OBJECT_GATEWAYS = "gateways";
+    private static final String OBJECT_META_DATA = "metadata";
+
+    @Override
+    public String getProviderName() {
+        return "theThingsStack";
```
Comment (sophokles73) [inline adapters/lora-vertx/src/main/java/org/eclipse/hono/adapter/lora/providers/TheThingsStackProvider.java:71]: why not `The Things Stack`?

**Review by sophokles73 — APPROVED** [review]
LGTM

File: adapters/lora-vertx/src/main/java/org/eclipse/hono/adapter/lora/providers/TheThingsStackProvider.java (hunk 1 above)
Comment (BobClaerhout) [reply adapters/lora-vertx/src/main/java/org/eclipse/hono/adapter/lora/providers/TheThingsStackProvider.java:71]: I don't really have a preference but it's the similar to `actiltyWireless` and `actilityEnterprise`.

Comment (sophokles73) [reply adapters/lora-vertx/src/main/java/org/eclipse/hono/adapter/lora/providers/TheThingsStackProvider.java:71]: Ok, then let's stick with it

**Comment by sophokles73** [comment]
@BobClaerhout would you mind also adding a corresponding entry to the release notes?

**Comment by BobClaerhout** [comment]
> @BobClaerhout would you mind also adding a corresponding entry to the release notes?

Sure: https://github.com/eclipse/hono/pull/2527

**Pull request merged**